Xwt.XamMac/Xwt.Mac/TreeViewBackend.cs

```diff
@@ -192,6 +192,30 @@ public void ExpandToRow (TreePosition pos)
 				p = source.GetParent (p);
 			}
 		}
+
+		public TreePosition GetRowAtPosition (Point p)
+		{
+			var row = Table.GetRow (new System.Drawing.PointF ((float)p.X, (float)p.Y));
+			return row >= 0 ? ((TreeItem)Tree.ItemAtRow (row)).Position : null;
+		}
+
+		public Rectangle GetCellBounds (TreePosition pos, CellView cell, bool includeMargin)
+		{
+			var it = tsource.GetItem (pos);
+			if (it == null)
+				return Rectangle.Zero;
+			var row = Tree.RowForItem (it);
+			return GetCellBounds (row, cell, includeMargin);
+		}
+
+		public Rectangle GetRowBounds (TreePosition pos, bool includeMargin)
+		{
+			var it = tsource.GetItem (pos);
+			if (it == null)
+				return Rectangle.Zero;
+			var row = Tree.RowForItem (it);
```

Comment:
Ok, and what about other RowForItem calls ([Line 148](https://github.com/mono/xwt/pull/433/files#diff-3e118f34da2c8971c2c95f0591537432R148) or [Line 154](https://github.com/mono/xwt/pull/433/files#diff-3e118f34da2c8971c2c95f0591537432R154))? Shouldn't it always be casted then?